subsys/bluetooth/host/gatt.c

```diff
@@ -2470,10 +2471,16 @@ static void gatt_find_info_rsp(struct bt_conn *conn, u8_t err,
 			 * entry must be its value.
 			 */
 			if (!bt_uuid_cmp(&u.uuid, BT_UUID_GATT_CHRC)) {
-				if (length >= len) {
-					pdu = (const u8_t *)pdu + len;
-					length -= len;
+				i--;
+				/* If value is the last one on the response
+				 * exit the loop and adjust the handle.
+				 */
+				if (i == 1) {
+					i--;
+					handle += 1;
```

Comment:
How come you're mixing two different increment/decrement conventions on consecutive lines? Why not just handle++?

Reply:
I forgot push the lastest version so the code no longer tries to skip in place and treat the corner case where it is the last attribute, instead Im now just using a flag to tell if the next handle needs to be skipped which is more in line with the spec allowing gaps so we don't make the assumption it is handle++ but parse the handle and only then skip.